### DISCUSSION TITLE
Vimeo: private videos don't work

### DISCUSSION THREAD

**Discussion by smatlanta** [aclap-dev/video-downloadhelper#42] [Help and Q&A] [answered]
### Your question

I have a paid-for "lesson site" on Vimeo
For years I've downloaded my video lessons with no problem but now VDH shows no media recognized.  I have also updated my MacOS to Sonoma and use VDH version: 8.2.0.18a1
channel: beta
build date: 2024-02-01

Firefox

Had anything changed that make me unable to used VDH anymore for this purpose?

**Comment by paulrouget** [comment]
Yes things have changed.

Can you share a link of a failing video ?

**Comment by smatlanta** [comment]
Thanks so much for the fast response!  (I didn't expect that)

I've always loved using VDH, and have used it for years for this purpose
(my lessons).  I hope I can still use VDH

Here is the link... but it's subscriber-only so is there any way I can help
you help me?

steve

**Comment by paulrouget** [comment]
> it's subscriber-only

Ah, this is tricky.

But if it used to work, I think I know what's going on (note to self: this probably requires the JSON MPD probe).

I won't have time to fix this for 8.2. So stay on 8.1 (disable autoupdates). And I hope i'll fix that by 8.3.

**Comment by smatlanta** [comment]
Ha! I'm not sure what you are saying about the JSON MPD probe, But does
that mean you will get back to me?  Thanks.

I can give you a password if that's what you will need to fix it.

I appreciate your help.  This is the main reason I have VDH so I hope
you'll be able to help get it back to working

thanks,
steve

**Comment by paulrouget** [comment]
> But does that mean you will get back to me? 

Yes. I'll need a little bit of time to fix it. Hopefully in v8.3. I'll comment here once it's done.

**Comment by smatlanta** [comment]
Yay! Thanks so much for the great service :)

**Comment by paulrouget** [comment]
Getting more report about this: https://github.com/aclap-dev/video-downloadhelper/discussions/48

**Comment by smatlanta** [comment]
Great! I'm glad it's not just a "one-person" thing.  It seems to be
affecting other Vimeo users.  I have another guitar lesson site that I also
pay for.  It's NOT Vimeo and its working fine.

Just think how many people are going to LOVE you when you figure out how to
fix it!

Keep in touch...

thanks again,
steve

**Comment by GrampaWildWilly** [comment]
We, meaning fellow users, have already figured out how to get Vimeo videos.  It's back on the old Google support forum.  Look for a thread with the words "too fast" in the title.  A search on "too fast" will turn up many hits, but most of those are references to the thread you want.  You need to find the thread with "too fast" in the title.  Your solution is in there if you're patient enough to learn what we teach in there.  That will get you through until Paul comes up with the solution.

**Comment by smatlanta** [comment]
Thanks.

I guess I better wait for Paul.  Thank you for trying to help me but I am
not a computer coder or programmer.  I love using VDH and when it stopped
working I looked for "support" and the next thing I find out is I'm part of
a group of computer-coding-programming geniuses! (I can't even spell it!
LOL)    I tried plugging "too fast" into some of the discussion search
engines but I must be doing something wrong because I am not getting an
output of a bunch of titles with "too fast" in the discussion.  I didn't
know how to get to "the old Google support forum" If there is a link that I
can click on and follow the directions of the answer, then I could
probably do it from there.  If not, I will have to wait for Paul to
hopefully fix it for us.

Thanks again for trying to help.

**Reply by GrampaWildWilly** [reply]
https://groups.google.com/g/video-downloadhelper-q-and-a

Search in there, not on Google in general.  There's a search field at the top of that page.  Put the search words in " " to get more relevant results.

**Reply by GrampaWildWilly** [reply]
And you don't have to be a genius to write computer programs.  It's just a skill you can learn like any other, like knitting or ballet or singing or tennis or jigsaw puzzles.  Remember.  A computer is just a junk pile of electronic components.  It's a tool, like a wrench or pliers.  Doesn't take a genius to learn how to program one.  You learned how to use one.  That's just another skill.  Don't be intimidated.

**Reply by QuinGold** [reply]
@smatlanta - fwiw, I couldn't figure out the solution in the old forum to fix the "fast" videos (where the video runs at 12,000 fps). But I worked around it by downloading the video and audio separately, fixing the frame rate in Adobe Premiere, and re-rendering. I would love if 8.3 fixes this. Not all videos download "fast".  I've turned off auto updating in Firefox though as 8.2 is not the answer.

**Comment by paulrouget** [comment]
I'm trying to reproduce the issue here, but i'm having difficulties.

This is what I do:
- i load a website that has a private vimeo video
- i press the play button of the private vimeo
- i see the VDH button light up
- i click on the VDH button, and it shows 3 potential downloads
- i click on the first entry, and it downloads the video fully

@smatlanta @QuinGold - please let me know at which steps things get different for you?

Also, if that's possible, can you share the link to the failing video? Maybe I'll create an account to give it a test.

**Reply by QuinGold** [reply]
[edited] I will message you a link (I don't want to publish it here).

**Comment by smatlanta** [comment]
The link is


https://barrygreenevideolessons.vhx.tv/videos/the-anatomy-of-a-compelling-jazz-line

It's technically VHXtv which is owned by Vimeo.

You have to pay for the site.  By the way, if you are a jazz guitar
player... this is an awesome site and probably the best jazz guitar lessons
out there!

It has always worked before ... until the last 30 days

**Reply by QuinGold** [reply]
@smatlanta - I just reinstalled the beta (Feb 8) and now my movies are all downloading on Mac with just Quick Download. They don't shudder (no need to Convert like before) and the speed is normal. I'm cautiously ecstatic... :)

**Reply by paulrouget** [reply]
@smatlanta thank you - I can reproduce your issue I believe. Looking into it!

**Comment by smatlanta** [comment]
*** Late Breaking****

Thanks. I tried it by trying it in Chrome and it works! (?)

The only problem (at least for me) is it says my VDH Conversion License is
only for FireFox so I would have to buy another one for Chrome.  I will do
that if I have to, but if you can fix Firefox, that would be great.  Maybe
the fact that it works on Chrome will help you fix it (?)

**Comment by smatlanta** [comment]
I'll try that.  Thanks.   I'm cautiously ecstatic... also   LOL

And at least I know it works in Chrome if I buy an additional license
that's only for Chrome (otherwise there is a watermark)

**Reply by QuinGold** [reply]
I had bought both Firefox and Chrome, but found Firefox worked better actually. Plus it works in YouTube on Firefox.

**Comment by paulrouget** [comment]
From what I'm understanding, this actually works as long as we don't use Download & Convert. With v8.2.

So I'm assuming this is solved?

If not, please let me know.

**Comment by QuinGold** [comment]
Seems like something got fixed overnight!

I just tried it on a Thinkific site and it worked fine. If I did Quick Download, I got an .mkv file so it’s nice to be able to Convert to .mov or .mp4.

On a Zenler/Vimeo site:
If I Download & Convert to an Re-encoded MP4, then it stops at the Converting stage with a red error. However, the movie is fine (31 MB) on the hard drive.

If I Download & Convert to a .mov, it moved to Converting with no error (movie is 22MB, which is odd as it’s normally bigger than Re-encoded MP4). 

Maybe it doesn’t like re-encoding an .mp4 to an .mp4? 

PS. It’s so great to not have to waste all day fixing fast movies!

**Reply by BrendaW2024** [reply]
Hi

I've been following this thread and hoping my challenges will be resolved. I'm still not able to download videos that are behind passwords. I can finally give an example of a site that isn't behind a paywall but does require you sign up for the programme if you're interested in exploring - I believe the replays are free for the weekend.

[Site](https://www.consciouslife.com/conferences/tsc-4/)

I've signed up to the beta channel of Firefox and Chrome. On both browsers, downloads start but don't complete. They're now getting stuck at various %s - not just 9%. The furthest I've seen a file go so far is 66% - I tried all 3 variants for a particular video to get this result. 

 Here are details - Firefox:

version: 8.2.0.19a1
channel: beta
build date: 2024-02-08
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-GB
license: accepted
key (only 16 first characters):xxxx
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.15",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/Users/xxxx",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "n6.1-9-g63b936e978"
}

Chrome:

version: 8.2.0.19
channel: beta
build date: 2024-02-08
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-GB
license: accepted
key (only 16 first characters): xxxx
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.15",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/Users/xxxx",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "n6.1-9-g63b936e978"
}

**Reply by GrampaWildWilly** [reply]
Just to get the details out of the way, I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3085, Firefox 122.0.1 64-bit, licensed VDH 8.2.0.19a1 beta, CoApp 2.0.17.

I'm not going to sign up for a site that I will probably never visit again, even though it's free.  I did go to the page whose URL you provided.  I scrolled it down & discovered what appears to be a short promo for the site.  Without clicking the play button on it, VDH was showing me this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/a83db6ae-f8ef-4533-925f-fd1b4eaf070e)

So I downloaded the 1920x1080 variant at the top of the VDH menu:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/48757958-ae98-4c6e-b69d-9c61f8bbcc20)

Here are the results of the download:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f0c35b2f-dfe1-4e4c-8cc9-407e4c5ed603)

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f3352519-166a-4ddf-b0e8-74ed976a4e81)

It played fine in VLC, although I just skimmed it to make sure it was all there, video & audio, right to the end.

This may or may not be a good test since it's pretty short & unlikely to stall.  This test at least shows that the embedded Vimeo content on this site is something that VDH can now download.  This did not used to be true.  VDH has come a long way in a short time under the care of Paul.  He's writing this code.  I, on the other hand, don't work for VDH.  I'm just another user.  So it is encouraging to see that VDH is getting this content now.

Like I said, I am not signing up for this site even though it is free.  This is not content that interests me.  Paul may be motivated to sign up just to get some test cases.  Obviously, I can't speak for him & I'm not even cajoling him to sign up.  He'll make his own decision here.

In my experience, being behind a password is not something that has deterred VDH.  I don't think the password thing is the cause of the problem.  I think that's a coincidence.  I have used VDH behind a paywall without problems for several years.  I think the issue is something to do with the format of the content, not the fact that it's behind a password.  It would sure simplify problem diagnosis if we could get some samples of this happening that are not encumbered by either passwords or geoblocking.

**Reply by paulrouget** [reply]
@BrendaW2024 I registered and found the videos ([that one](https://www.consciouslife.com/conferences/tsc-4/sessions/unbroken-understanding-trauma-responses)). Ran the download and it reached 100% without any issue. Can you confirm that specific video is failing for you?

**Reply by GrampaWildWilly** [reply]
I didn't register but I clicked your link, Paul, anyway.  And it showed me the video.  A bit unexpected but certainly welcome.  The VDH menu remained empty until I clicked the play button & let it play for not even 1 second before I paused it.  That caused the VDH menu to populate.  I downloaded the 1920x1080 mp4 variant.  It took 4:03 to download a 929M mp4, duration 53:09.  It played fine in VLC, although I just skimmed it, sampling it at 1-minute intervals, & letting the last 15 seconds play uninterrupted.  I had video & audio at each sample point, including all the way to the end.  This one did not delay for 2-4 minutes then resume, like some other recent cases.  It just started, downloaded to completion, & gave me a good video.  This is on Windows 11 Pro 64-bit version 23H2 build 22631.3085, Firefox 122.0.1 64-bit, licensed VDH 8.2.0.19a1 beta, CoApp 2.0.17.

**Reply by BrendaW2024** [reply]
@paulrouget Thanks for being willing to explore. I was able to download the video in your message in Chrome, using the standard VDH, not the Beta version. I tried both, but the Beta version did not complete any of the variants that were available. 

I also tried a random selection of other videos using Chrome. I didn't always need to play the videos to see variants popping up as per @GrampaWildWilly's comments (thanks for your experimenting too).  

All the videos I tried completed downloading using standard VDH and the 640 x 360 variant. Most of the videos were perfect. Two downloaded as 'fast' videos ie the audio is ok but the picture is all scrunched up at the beginning of the video and not aligned with the audio track. These were [this video](https://www.consciouslife.com/conferences/tsc-4/sessions/how-to-not-traumatize-your-kids) and [this video](https://www.consciouslife.com/conferences/tsc-4/sessions/how-trauma-becomes-conditioned). I tried other variants of these videos too - same 'fast' video result. 

I tried using the Beta VDH for a couple of the videos I did download using standard vdh - none completed. They didn't seem to get to a standard place - some bombed out early, others a little further along. When I aborted these videos, I got a side download error. 

Then I thought I'd try my luck on a Kajabi (password-protected, behind a paywall) site - neither VDH version completed any downloads. 

Hope that all makes sense.

**Reply by GrampaWildWilly** [reply]
I downloaded the 2 videos you say were "scrunched up" & for me, they were fine.  I'm looking at your environment & comparing it to mine.  I see two differences.  You're using CoApp 2.0.15 & I'm on 2.0.17.  I'm not sure just how much is different between those 2.  Paul could say.  The differences may or may not be relevant to what you're observing.  The other difference is you are on Mac & I'm on Windows.  I can't even begin to guess what difference that makes.  I've never used Mac so I have no idea how it works.  Maybe Paul can say how that might influence the results here.

The downloads that get too fast video with perfectly fine audio is something we discussed back on our old Google forum going back well over a year.  We came up with alternative methods of downloading such content.  Those techniques used ffmpeg instead of VDH to download the videos.  But that was before VDH itself started downloading content with ffmpeg.  My experience has been that this too fast video problem has gone away.

It suddenly occurs to me to ask what you have in your VDH settings under the Behavior tab for the download processor.  If you are not using the CoApp, your downloads won't use ffmpeg & there's a chance you wouldn't gain the benefit of the CoApp.  Installing the CoApp by itself does not make VDH use the CoApp (meaning ffmpeg) for downloading.  You have to go into the VDH settings & tell VDH to use the CoApp.  Using the CoApp is NOT the default VDH behavior.  Strange but true.  I think we've been assuming that you are actually using the CoApp but we haven't explicitly eliminated that possibility.  What do you have for this?

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/32a4fbe5-985d-4b59-affa-b0022ffdfae0)

**Reply by BrendaW2024** [reply]
Thanks for the heads up about the updated CoApp - I didn't know there was a new version. I updated that then tried the 2 videos again using the Beta VDH. One downloaded fine and one didn't complete. I tried the one that didn't complete using the standard vdh again - I tried all the available variants and they were all 'fast'. (I have figured out how to fix this btw using a combination of some of the comments from the old group that I could understand and a bit of help from chat GPT!!)

Then I checked my Behaviour settings as you suggested - both standard VDH and Beta VDH were set to 'Ask'. 

<img width="452" alt="Behaviour" src="https://github.com/aclap-dev/video-downloadhelper/assets/159222797/839d6054-379d-4ad1-b692-8987d0b8b7d9">

I changed that to Companion Application and tried the file again. All the variants using standard VDH downloaded but the video track was compressed at the beginning again and not aligned with the audio. Only 1 variant was available using Beta VDH - this time the file got stuck at 76%. When I aborted the download I got the same side download error as before. 

I tried a couple of other files using standard VDH. Some downloaded fine, some had the misaligned video and audio problem but I couldn't figure out what the difference was or why this would be happening - I wonder if it had something to do with the format the files were uploaded in.

**Reply by paulrouget** [reply]
Please follow these instructions and share the informations about your settings: https://github.com/aclap-dev/video-downloadhelper/wiki/Where-can-I-find-my-versions%3F

**Reply by BrendaW2024** [reply]
On Chrome, which is what I've been using during this conversation, I have both standard VDH and the Beta version installed.

This is the info for standard VDH:

version: 8.1.6.0
beta: false
build date: 2024-01-18
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-GB
license: accepted
key (only first 16 fisrt characters): xxxx
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.17",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/Users/xxxx",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "n6.1-9-g63b936e978"
}

This is from the Beta version:

version: 8.2.0.19
channel: beta
build date: 2024-02-08
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-GB
license: accepted
key (only 16 first characters): xxxx
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.17",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/Users/xxxx",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "n6.1-9-g63b936e978"
}

**Comment by smatlanta** [comment]
Well, honestly, I'm a little lost in all this. Sorry.

It works in Chrome but still does not work Firefox.... at least doing it
the normal way, which for me is opening a video, going to the top of my
display and click what SHOULD be a colored icon (if it's capturing media
successfully) In my case it stays gray. I open and play the video and No
Media To Process in the Current Tab comes up.

It does work on other stuff... but NOT on the VHX.tv stuff.   (It DOES work
normal in Chrome)

Am I supposed to use a different version of VDH right now for Firefox?
Will an older version work for now?  If so, how do I download an older
version?

I am currently using:
Video DownloadHelper
Version 8.2.0.19a1 (beta)
Browser locale: en-US
Production build
Built on 2024-02-08
Build options: browser=firefox, linuxlic=false, noyt=false, target=mozilla

**Comment by smatlanta** [comment, marked as answer]
Well, as it turns out, this latest companion update must have fixed the
issue I was having because everything works great now *VdhCoApp 2.0.17*

*Thanks for all the support!*
*Steve*
